sharkfuser/include/fusilli/node/conv_node.h

```diff
@@ -338,6 +339,26 @@ class ConvWGradNode : public NodeCRTP<ConvWGradNode> {
                             "Tensor '" + xT->getName() +
                                 "' is neither contiguous nor channels-last as "
                                 "defined by its stride");
+
+    // Group count checks
+    constexpr size_t channelsIdx = 1;
+    int64_t inChannels = xT->getDim()[channelsIdx];
+    int64_t outChannels = dyT->getDim()[channelsIdx];
+    int64_t filterChannels = dwT->getDim()[channelsIdx];
+    FUSILLI_RETURN_ERROR_IF(
+        inChannels % filterChannels != 0, ErrorCode::InvalidAttribute,
+        "Conv input channels must be divisible by the filter channels");
+
+    int64_t groupCount = inChannels / filterChannels;
+    FUSILLI_RETURN_ERROR_IF(
+        groupCount <= 0 || groupCount > inChannels || groupCount > outChannels,
+        ErrorCode::InvalidAttribute,
+        "Conv group count must be greater than 0 and less than or equal to the "
+        "numbers of input and outputs channels");
+    FUSILLI_RETURN_ERROR_IF(
+        outChannels % groupCount != 0, ErrorCode::InvalidAttribute,
+        "Conv output channels must be divisible by the group count");
```

Comment:
This may be confusing to interpret. Consider being explicit here - like

```suggestion
        "ConvWGrad output (DY) channels must be divisible by the group count");
```